jaeger-core/src/main/java/io/jaegertracing/internal/samplers/PerOperationSampler.java

```diff
@@ -64,6 +63,20 @@ public synchronized boolean update(OperationSamplingParameters strategies) {
       isUpdated = true;
     }
 
+    //remove the operation samplers that are not present in the strategies
+    Set<String> newOps = new HashSet<String>();
+    for (PerOperationSamplingParameters strategy : strategies.getPerOperationStrategies()) {
+      newOps.add(strategy.getOperation());
+    }
+
+    for (String op : operationNameToSampler.keySet()) {
+      if (!newOps.contains(op)) {
+        operationNameToSampler.remove(op);
+        isUpdated = true;
+      }
+    }
```

Comment:
Could it be simpler to create a new `HashMap<String, GuaranteedThroughputSampler>`, and either insert a sampler update, or create a new sampler; and replace `operationNameToSampler` with this new Map?

Reply:
You are right, I thought about recreate the remote sampler using the response and then thought we need to preserve the sampler's internal state, haven't thought about re-insert the sampler to new map. Updating.